torch/_dynamo/variables/misc.py

```diff
@@ -635,6 +635,29 @@ def call_method(
         return super().call_method(tx, name, args, kwargs)
 
 
+class MethodWrapperVariable(variables.ConstantVariable):
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        if self.value.__name__ == "__get__":
+            assert len(args) == 1 and len(kwargs) == 0
+            return args[0].var_getattr(tx, self.value.__self__.__name__)
+
+        super().call_function(tx, args, kwargs)
+
+
+class GetSetDescriptorVariable(variables.ConstantVariable):
+    def var_getattr(self, tx, name):
+        if name == "__get__":
+            from .builder import VariableBuilder
+
+            return VariableBuilder(tx, AttrSource(self.source, "__get__"))(
+                self.value.__get__
+            )
+        else:
+            return super().var_getattr(tx, name)
```

Comment:
So, I think the conclusion from the Workchat discussion is:

1. These definitely shouldn't be ConstantVariable (mea culpa!)
2. If the inner state on these is immutable and you want to pretend that you just have multiple different functions, you can just make these TorchVariables
3. Having them just being new VT subclasses is also OK

Reply:
I feel like TorchVariable is kind of getting messy, so I think I'm gonna go with 3